Update documents to reflect change to Microsoft Entra ID (#1665)
alan-turing-institute/data-safe-haven: docs/source/roles/system_manager/migrate_an_shm.md

See the {ref}`Safe Haven Management documentation <roles_deployer_deploy_shm>` for more details.

## 5. {{file_folder}} Ensure the Microsoft Entra domain is registered

![Powershell: a few minutes](https://img.shields.io/static/v1?style=for-the-badge&logo=powershell&label=local&color=blue&message=a%20few%20minutes) at {{file_folder}} `./deployment/safe_haven_management_environment/setup`

See the {ref}`Safe Haven Management documentation <roles_deployer_deploy_shm>` for more details.

```{note}

> You will need to use a Microsoft Entra global admin when the `AzureAD` module asks you to sign-in.

Reviewer: ```suggestion
You will need to use a Microsoft Entra ID global admin when the `AzureAD` module asks you to sign-in.
```

Author: I think this should be Microsoft Entra global admin, going off the guide I linked to:

> 
Azure AD admin| Microsoft Entra admin
-- | --

See the {ref}`Safe Haven Management documentation <roles_deployer_deploy_shm>` for more details.

```{note}

> You will need to use a Microsoft Entra global admin when the `AzureAD` module asks you to sign-in.

Reviewer: ```suggestion
You will need to use a Microsoft Entra ID global admin when the `AzureAD` module asks you to sign-in.
```

Author: Again, official guidance suggests no use of ID here

$userOuPath = (Get-ADObject -Filter * | Where-Object { $_.Name -eq "Safe Haven Research Users" }).DistinguishedName
$users = Get-ADUser -Filter * -SearchBase "$userOuPath" -Properties *

# Connect to Microsoft Entra ID

> # Use the credentials for a Microsoft Entra global admin (eg. `aad.admin.firstname.surname@<SHM domain>`)

Reviewer: ```suggestion
# Use the credentials for a Microsoft Entra ID global admin (eg. `aad.admin.firstname.surname@<SHM domain>`)
```

Author: Again, official guidance suggests no use of ID here

Connect-MsolService

> # Reset source anchor for Microsoft Entra users

Reviewer: ```suggestion
# Reset source anchor for Microsoft Entra ID users
```

Author: The guidance is not explicit, but given the things like `admin` and `tenant` are not preceded by `ID`, I'd think `users` shouldn't be either.

All research users in this SHM will have to go to `https://aka.ms/sspr` to reset their passwords although their MFA configuration will stay the same.
```

### {{train}} Install Microsoft Entra Connect

![Remote: ten minutes](https://img.shields.io/static/v1?style=for-the-badge&logo=microsoft-onedrive&label=remote&color=blue&message=ten%20minutes)

See the {ref}`Safe Haven Management documentation <roles_deployer_shm_aad_connect>` for more details.

````{note}

> Microsoft Entra Connect is the new name for Azure AD Connect. However, while all Microsoft documentation and entries in the Azure portal now refer to Microsoft Entra Connect, as of the release of `v4.2.0` of the Data Safe Haven, the software itself is still named `Azure Active Directory Connect`, and will appear as such on your Domain Controller. 

Reviewer: ```suggestion
Microsoft Entra Connect is the new name for Azure AD Connect. However, while all Microsoft documentation and entries in the Azure portal now refer to Microsoft Entra Connect, as of this release of Data Safe Haven, the software itself is still named `Azure Active Directory Connect`, and will appear as such on your Domain Controller. 
```

If this happens then you will need to wait for the previous disconnection to complete, which may take up to 72 hours.
````

### {{recycle}} Update Microsoft Entra Connect rules

![Remote: one minute](https://img.shields.io/static/v1?style=for-the-badge&logo=microsoft-onedrive&label=remote&color=blue&message=one%20minute)

See the {ref}`Safe Haven Management documentation <roles_system_deployer_shm_aad_connect_rules>` for more details.

> ### {{put_litter_in_its_place}} Unregister the old domain controller in Microsoft Entra

Reviewer: ```suggestion
### {{put_litter_in_its_place}} Unregister the old domain controller in Microsoft Entra ID
```
